app/controllers/api/v1/posts_controller.rb

```diff
@@ -7,30 +7,39 @@ class PostsController < Api::V1::BaseController
       before_action :find_post, only: %i[show update]
 
       def index
+        return render json: { message: 'User does not exist' }, status: :bad_request unless @user.present?
+
         @posts = @user.posts.order(published_at: :desc)
       end
 
-      def show; end
+      def show
+        render json: { message: 'Post does not exist' }, status: :bad_request unless @post.present?
+      end
 
       def create
+        return render json: { message: 'Unauthorized' }, status: :unauthorized unless @user.present?
```

Comment:
Let's do some code refactoring! You are both rendering Unauthorized here and in `validate_post`. You can do just one check and do something like:
```suggestion
        return render json: { message: 'Unauthorized' }, status: :unauthorized unless @user == current_user
```